internal/registry/git_test.go

```diff
@@ -0,0 +1,78 @@
+package registry_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/heroku/color"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	ilogging "github.com/buildpacks/pack/internal/logging"
+	"github.com/buildpacks/pack/internal/registry"
+	"github.com/buildpacks/pack/logging"
+	h "github.com/buildpacks/pack/testhelpers"
+)
+
+func TestGit(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+	spec.Run(t, "Git", testGit, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testGit(t *testing.T, when spec.G, it spec.S) {
+	var (
+		registryCache   registry.Cache
+		tmpDir          string
+		err             error
+		registryFixture string
+		outBuf          bytes.Buffer
+		logger          logging.Logger
+		username        string = "supra08"
+	)
+
+	bp := registry.Buildpack{
+		Namespace: "example",
+		Name:      "python",
+		Version:   "1.0.0",
+		Yanked:    false,
+		Address:   "example.com",
+	}
+
+	it.Before(func() {
+		logger = ilogging.NewLogWithWriters(&outBuf, &outBuf)
+
+		tmpDir, err = ioutil.TempDir("", "registry")
+		h.AssertNil(t, err)
+
+		registryFixture = h.CreateRegistryFixture(t, tmpDir, filepath.Join("..", "..", "testdata", "registry"))
+		registryCache, err = registry.NewRegistryCache(logger, tmpDir, registryFixture)
+		h.AssertNil(t, err)
+	})
+
+	it.After(func() {
+		err := os.RemoveAll(tmpDir)
+		h.AssertNil(t, err)
+	})
+
+	when("#createGitCommit", func() {
```

Comment:
Could you add a test where the RegistryCache wasn't created, to verify that it will call `CreateCache`? Or is that tested here already?

Reply:
Side tantrum: I have issues with the architecture decision. I'm not seeing the value of lazy loading the cache in this case. I think a better pattern may be to differentiate between a RegistryCache "config" and an "initialized" RegistryCache. Any time a `RegistryCache` is provided it should already be initialized. A `RegistryCacheConfig` can be used in cases where we just want to provide configuration without necessary interaction.

All this to say, I vote we omit this check if the hope is to change it soon.

Reply:
Issue (#818) created.